pkg/apis/app/v1alpha1/application_types.go

```diff
@@ -100,6 +101,34 @@ type InfoItem struct {
 
 	// Value is human readable content.
 	Value string `json:"value,omitempty"`
+
+	// Value is derived from another source.
+	ValueFrom *InfoItemSource `json:"valueFrom,omitempty"`
+}
+
+type InfoItemSource struct {
```

Comment:
InfoItemSource, ServiceSelector, and IngressSelector need comments.

Reply:
Done.